doc/design/game_design_document.md

```diff
@@ -0,0 +1,165 @@
+# Game design document: Project OMG
+
+## Introduction
+
+Project OMG is a 2D top-down, multiplayer, match-based game where players
+engage in real-time combat with intelligent bots in interactive environments.
+Players will create characters with unique skills and experience synergy between
+different abilities to defeat their opponents. The game focuses on strategic
+combat, team coordination, and dynamic interactions within the game world.
+
+## Game overview
+
+### Game concept
+
+Project OMG combines strategic depth with fast-paced action in a multiplayer
+setting. Players will create characters with distinct classes and abilities,
+participating in matches where skill synergy plays a crucial role. The
+environment is interactive, with destructible objects that can influence
+gameplay. Matches are short, with no progression save, ensuring each game is a
+fresh experience.
+
+### Genre
+
+2D top-down, multiplayer, match-based combat.
+
+### Platform
+
+The game will be developed for PC initially, with potential expansion to include
+controller support and mobile platforms in the future.
+
+### Target audience
+
+The game is aimed at players aged 12 and above who enjoy strategy and
+multiplayer games, targeting both casual gamers and competitive players looking
+for an engaging multiplayer experience with a mix of PvP (Player vs Player)
+and PvE (Player vs Environment) elements.
+
+## Gameplay mechanics
+
+### Core mechanics
+
+- **Match-based gameplay**: Each game session is independent, with no
+  progression save. Players engage in short, intense matches.
+- **Character creation**: Players can create characters, choosing from various
+  classes and skills.
+- **Skill synergy**: Combining different skills (e.g., fire + poison) can create
+  unique effects and strategies.
+- **PvPvE**: Players compete against each other and AI-controlled bots, adding
+  an extra layer of challenge.
+- **Interactive environment**: The game world includes destructible objects and
+  other interactable elements that players can use to their advantage.
+
+### Game flow
+
+1. **Matchmaking**: Players join a lobby and are matched with others based on
```

Comment:
In core mechanics section, we claim that no progression will be saved. However, here, we propose to set up a match-making system based on skill levels, which requires saving progression. I think this creates a contradiction.

Reply:
You are right, fixed it now.